stratum/lib/security/certificate.cc

```diff
@@ -54,43 +63,42 @@ util::Status GenerateRSAKeyPair(EVP_PKEY* evp, int bits) {
   BIGNUM_ptr exp(BN_new(), BN_free);
   BN_set_word(exp.get(), RSA_F4);
   RSA* rsa = RSA_new();
-  CHECK_RETURN_IF_FALSE(RSA_generate_key_ex(rsa, bits, exp.get(), NULL))
+  CHECK_RETURN_IF_FALSE(RSA_generate_key_ex(rsa, bits, exp.get(), nullptr))
       << "Failed to generate RSA key.";
-
   // Store this keypair in evp
   // It will be freed when EVP is freed, so only free on failure.
   if (!EVP_PKEY_assign_RSA(evp, rsa)) {
     RSA_free(rsa);
     RETURN_ERROR(ERR_INVALID_PARAM) << "Failed to assign key.";
   }
+
   return util::OkStatus();
 }
 
 util::Status GenerateUnsignedCert(X509* unsigned_cert,
                                   EVP_PKEY* unsigned_cert_key,
                                   const std::string& common_name, int serial,
-                                  int days) {
+                                  absl::Time valid_until) {
   CHECK_RETURN_IF_FALSE(
       ASN1_INTEGER_set(X509_get_serialNumber(unsigned_cert), serial));
   CHECK_RETURN_IF_FALSE(X509_gmtime_adj(X509_get_notBefore(unsigned_cert), 0));
```

Comment:
```suggestion
  CHECK_RETURN_IF_FALSE(X509_gmtime_adj(X509_getm_notBefore(unsigned_cert), 0));
```

`X509_get_notBefore` is deprecated, use: `X509_getm_notBefore`